### PR TITLE
Add health check for ports 8099 and 9099

### PR DESCRIPTION
**Pull request by binodmx**
Fixes https://github.com/wso2/product-apim/issues/11961

### CODE REVIEW

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java [hunk 1]
```diff
@@ -149,10 +154,11 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
     @Override
     public void channelInactive(ChannelHandlerContext ctx) throws Exception {
         String endpointName = WebsocketEndpointManager.getInstance().getEndpointName(port, tenantDomain);
-        if (endpointName == null) {
+        if (endpointName == null && !isHealthCheckCall) {
```
Comment (sajinieKavindya) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java:157]: If this is a health check request, do we need to perform any of this inside this method? I think we can skip whatever we do in this method and return for a health check request. WDYT?

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java [hunk 2]
```diff
@@ -169,6 +175,17 @@ private void handleHandshake(ChannelHandlerContext ctx, FullHttpRequest req) thr
         if (log.isDebugEnabled()) {
             WebsocketLogUtil.printHeaders(log, req, ctx);
         }
+
+        // This block is for the health check of the ports 8099 and 9099
+        if (!req.headers().contains(HttpHeaders.UPGRADE) && req.uri().equals("/health")) {
+            isHealthCheckCall = true;
+            FullHttpResponse httpResponse = new DefaultFullHttpResponse(HttpVersion.HTTP_1_1, HttpResponseStatus.OK);
+            httpResponse.headers().set("content-type", "text/plain; charset=UTF-8");
```
Comment (sajinieKavindya) [inline components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java:183]: Any reason for setting the content-type and content-length headers here? I do not see any content adding to the FullHttpResponse here. Maybe we can skip setting these headers unless content is supposed to be sent along with this response. WDYT?

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java (hunk 1 above)
Comment (binodmx) [reply components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java]: Added a condition to check isHealthCheckCall at top of the method.

File: components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java (hunk 2 above)
Comment (binodmx) [reply components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/websocket/InboundWebsocketSourceHandler.java]: Removed the headers.

**Review by sajinieKavindya — DISMISSED** [review]

**Review by Arshardh — DISMISSED** [review]

**Review by Arshardh — APPROVED** [review]

**Pull request merged**